api/src/api/emailCheckRouter.ts

```diff
@@ -0,0 +1,34 @@
+import { UserDataClient } from "@domain/types";
+import type { LogWriterType } from "@libs/logWriter";
+import { Router } from "express";
+import { StatusCodes } from "http-status-codes";
+
+export const emailCheckRouterFactory = (userDataClient: UserDataClient, logWriter: LogWriterType): Router => {
+  const router = Router();
+
+  router.post("/emailCheck", async (req, res) => {
+    const { email } = req.body;
+
+    const logId = logWriter.GetId();
+    let status;
+
+    if (email === undefined) {
+      status = StatusCodes.BAD_REQUEST;
+      res.status(status).send({ error: "`email` property required." });
+    } else {
+      const userData = await userDataClient.findByEmail(email);
+
+      if (userData) {
+        status = StatusCodes.OK;
+        res.status(status).send({ email, found: true });
+      } else {
+        status = StatusCodes.NOT_FOUND;
+        res.status(status).send({ email, found: false });
+      }
```

Comment:
Just kindof a general thought, do we need a response body or is the status code enough for us to work with on the frontend?